### ISSUE TITLE
Implement error handling for pricing flows

### ISSUE THREAD

**Issue by heyjay44** [web3-storage/web3.storage#2118]
### Problem Statement
When customers enter their payment details, if there are issues with the details, they are not flagged in plain english. Various error codes are populated but they don't allow a user to understand where a mistake was made and therefore how to rectify.

### Proposed Work
Implement reasonable error handling coverage for the common pricing flows.  As part of this, review common stripe error codes returned and handle appropriately (particularly ones surrounding cc entry and subscription signups). 

### Success Criteria
When there is an issue with payment details, the user can see a plain english translation of the error.

**Comment by heyjay44** [comment]
Issue closed since most unexpected errors are captured already and a coherent, user-friendly message is shared on the UI.

**Comment by gobengo** [comment]
* a place where there is error message rendering around payment method form: https://github.com/web3-storage/web3.storage/blob/main/packages/website/components/account/addPaymentMethodForm/addPaymentMethodForm.js#L82
  * `defaultErrorMessageForEndUser` error message text: https://github.com/web3-storage/web3.storage/blob/main/packages/website/lib/api.js#L313
